lost/src/main/java/com/mapzen/android/lost/internal/LocationEngine.java

```diff
@@ -35,21 +35,37 @@ public LocationEngine(Context context, Callback callback) {
   public abstract boolean isProviderEnabled(String provider);
 
   /**
-   * Enables the engine on receiving a valid location request. Disables the engine on receiving a
-   * {@code null} request.
+   * Enables the engine on receiving a valid location request.
    *
-   * @param request Valid location request to enable or {@code null} to disable.
+   * @param request Valid location request to enable.
    */
-  public void setRequest(LocationRequest request) {
+  public void addRequest(LocationRequest request) {
     if (request != null) {
       this.request.addRequest(request);
       enable();
-    } else {
-      this.request.removeAllRequests();
+    }
+  }
+
+  /**
+   * Disables the engine when no requests remain, otherwise updates the engine's configuration.
+   *
+   * @param request Valid location request to enable.
+   */
+  public void removeRequest(LocationRequest request) {
+    if (request != null) {
+      this.request.removeRequest(request);
       disable();
+      if (!this.request.getRequests().isEmpty()) {
+        enable();
+      }
     }
   }
 
+  public void removeAllRequests() {
```

Comment:
As a possible future enhancement, should this also be invoked when the last client is disconnected to avoid this issue of needing to explicitly remove all requests before shutting down a client?